ReactSkia/components/RSkComponentImage.cpp

```diff
@@ -268,25 +268,28 @@ void RSkComponentImage::requestNetworkImageData(ImageSource source) {
 
 
   // completioncallback lambda fuction
-  auto completionCallback =  [&](void* curlresponseData,void *userdata)->bool {
+  auto completionCallback =  [this, remoteCurlRequest](void* curlresponseData,void *userdata)->bool {
     CurlResponse *responseData =  (CurlResponse *)curlresponseData;
     CurlRequest * curlRequest = (CurlRequest *) userdata;
     if((!responseData
         || !processImageData(responseData->responseurl,responseData->responseBuffer,responseData->contentSize)) && (hasToTriggerEvent_)) {
       sendErrorEvents();
     }
-    delete curlRequest;
+    //Reset the lamda callback so that curlRequest shared pointer dereffered from the lamda 
+    // and gets auto destructored after the completion callback.
+    remoteCurlRequest->curldelegator.CURLNetworkingHeaderCallback = nullptr;
```

Comment:
instead of resetiing this function pointer in its call , can it be done in sdk itself after the function is called ?